pytest_plugins/rerun_rp/rerun_rp.py

```diff
@@ -100,58 +58,108 @@ def pytest_addoption(parser):
+
+        Usage: --rp-reference-launch-uuid [value]
+
+        Value: report_portal_launch_uuid
+    '''
+    parser.addoption("--rp-reference-launch-uuid", nargs='?', help=help_text)
+    help_text = '''
+        Uses given satellite version + snap as a reference for parsing RP tests to be used with
+        --only-failed, --only-skipped or --user
+        e.g 6.10.0-1.0
 
-        Usage: --upgrades-rerun
+        Usage: --sat-version [value]
+
+        Value: sat_version
     '''
-    parser.addoption("--upgrades-rerun", action='store_true', default=False, help=help_text)
+    parser.addoption("--sat-version", help=help_text)
 
 
 @pytest.hookimpl(tryfirst=True)
 def pytest_collection_modifyitems(items, config):
     """
-    Collects and modifies tests collection based on pytest options to select tests marked as
-    failed/skipped and user specific tests in Report Portal
+    Collects and modifies test collection based on the pytest options to select the tests marked as
+    failed/skipped and user-specific tests in Report Portal
     """
+    rp_url = settings.report_portal.portal_url or config.getini('rp_endpoint')
+    rp_uuid = getenv('RP_UUID') or config.getini('rp_uuid') or settings.report_portal.api_key
+    rp_launch_name = (
+        getenv('RP_LAUNCH') or settings.report_portal.launch_name or config.getini('rp_launch')
+    )
+    rp_project = config.getini('rp_project') or settings.report_portal.project
     fail_args = config.getoption('only_failed', False)
     skip_arg = config.getoption('only_skipped', False)
     user_arg = config.getoption('user', False)
-    upgrades_rerun = config.getoption('upgrades_rerun', False)
-    if not any([fail_args, skip_arg, user_arg, upgrades_rerun]):
+    sat_version = config.getoption('sat_version', None)
+    ref_launch_uuid = config.getoption('rp_reference_launch_uuid', None) or config.getoption(
+        'rp_rerun_of', None
+    )
+    tests = []
+    if not any([fail_args, skip_arg, user_arg]):
         return
-    rp = ReportPortal()
-    version = settings.server.version
-    sat_version = f'{version.base_version}.{version.epoch}'
-    logger.info(f'Fetching Report Portal launches for target Satellite version: {sat_version}')
-    launch = next(
-        iter(
-            rp.launches(
-                sat_version=sat_version, launch_type='upgrades' if upgrades_rerun else 'satellite6'
-            ).values()
+    rp = ReportPortal(rp_url=rp_url, rp_uuid=rp_uuid, rp_project=rp_project)
+
+    if ref_launch_uuid:
+        logger.info(f'Fetching A reference Report Portal launch {ref_launch_uuid}')
+        try:
+            ref_launches = rp.get_launches(uuid=ref_launch_uuid)
+        except IndexError:
+            raise pytest.UsageError(
+                f'Provided reference launch {ref_launch_uuid} was not found or is not finished'
+            )
+    else:
+        if not sat_version:
+            sat_release = getenv('SAT_VERSION') or settings.server.version.get('release', '')
```

Comment:
Why prefer these two (seemingly unique) environment variables, when `settings.server.version.*` values can be provided via dynaconf envvars?

Reply:
I didn't realize that the dynaconf env vars are a thing, sorry.

Reply:
I'd remove it if you didn't have a specific plan for it. We really want to drive the pattern of using dynaconf settings + envvars to control things at runtime - and to consolidate on reading satellite version from settings. 

@JacobCallahan actually wrote a helper method to enforce a priority of resolving the version of the currently targeted satellite instance:
https://github.com/SatelliteQE/robottelo/blob/fb818ec0ed8f5bef823d107c78b8aed7f18e7140/robottelo/host_info.py#L156

It uses the Satellite class constructor which will resolve settings.server.hostname and get its version property, and if that fails will read `settings.server.version.release`, and then `settings.robottelo.satellite_version`, and then finally looks at `robottelo.constants.SATELLITE_VERSION` if there isn't a target satellite instance and `server.yaml` or `robottelo.yaml` don't have fields set.

Reply:
Agree with @mshriver here, we really need not to explicitly have our own Env var for satellite version when DynaConf provides that ability.

If other Env vars in this PR are for such a reason let's remove them.

Reply:
@rplevka looks like this still needs an update. I think you should use `robottelo.host.get_sat_version` (unstable branch location, its in host_info.py in master)

For snap number, we need to make sure we're setting `ROBOTTELO_SERVER__RELEASE__SNAP` in CI pipelines.

I also don't think we need an additional option (`sat_version`) for it on the CLI - just always use get_sat_version and `settings.server.release.snap`.

Reply:
@rplevka everything else in this PR looks good, and I was hoping to merge, but I don't want to include use of this env var and include a pattern that we don't want to perpetuate. Bumping to draft state so I know when to re-review.

Reply:
apparently we've got some float to string issue:
```
(Pdb++) from robottelo.host_info import get_sat_version
(Pdb++) get_sat_version
<function get_sat_version at 0x7efd3390ec10>
(Pdb++) get_sat_version()
<Version('6.1')>
```

Reply:
@mshriver testing the code with `robottelo.hosts.get_sat_version` revealed, that since this (plugin) code executes way before `align_to_satellite` fixture does - and `get_sat_version` relies on that while determining the full version from the host inventory.
This fact makes it to fall back sourcing the version from the settings, and if the setting is not set, it falls back to the `SATELLITE_VERSION` constant as a last resort - this last resort is a no-go for us, as it never contains a `.z` bit of the version (for apparent reasons). and it always defaults to `0`.

This would not work for our `z-branch` rerun pipelines.

I suggest strictly passing the version in a form of dynaconf env var in our pipelines, same as we do with the snap version.

Reply:
Updated the code so it works with

```
$ ROBOTTELO_SERVER__VERSION__RELEASE=6.10.0 ROBOTTELO_SERVER__VERSION__SNAP=18.0 pytest  tests/foreman/api/test_operatingsystem.py  --collect-only --only-failed
```
yet, fails with `ROBOTTELO_SERVER__VERSION__RELEASE=6.10`.

Reply:
ack, having `settings.server.version.release set to the full x.y.z string in CI is something we can and should do. We already calculate it based on what we get back from the satellite instance checkout.

In the case of this hook - if it doesn't have a full version, just skip the collection modification. If you want to select only failed cases, you have to be specific about the reference version.